### DISCUSSION TITLE
Object Tracking after the Detection

### DISCUSSION THREAD

**Discussion by amananand97** [ultralytics/yolov5#5351] [Q&A] [answered]
I am using an IP camera and controlling it with Onvif protocol. So is there a way to track an object just after the the YOLOv5 detection or using the trained weights to track an object automatically through the camera feed?
I'm just a beginner in computer vision, if there's any other documentation it would be really helpful.

**Comment by glenn-jocher** [comment, marked as answer]
@amananand97 👋 Hello! Thanks for asking about object **tracking** in computer vision. YOLOv5 🚀 is an object **detector** that detect, localizes and classifies objects in a single image. It does not connect objects across multiple images, for this you need a tracking solution. A few possible tracking solutions are:

- **Extended Kalman Filter (EKF)**: https://en.wikipedia.org/wiki/Kalman_filter
- **KLT tracker**: https://en.wikipedia.org/wiki/Kanade%E2%80%93Lucas%E2%80%93Tomasi_feature_tracker
- **YOLOv5 DeepSort tracker**: https://github.com/mikel-brostrom/Yolov5_DeepSort_Pytorch

Good luck and let us know if you have any other questions!
